yew/src/virtual_dom/mod.rs

```diff
@@ -176,60 +176,78 @@ enum Patch<ID, T> {
     Remove(ID),
 }
 
-/// Reform of a node.
-enum Reform {
-    /// Don't create a NEW reference (js Node).
-    ///
-    /// The reference _may still be mutated_.
-    Keep,
+/// Delay detachment of ancestors until after inserting new nodes
+struct DelayDetach<'a> {
```

Comment:
This allows us to insert a new node before the ancestor before detaching it. This way, we don't need `detach()` to return the `next_sibling` (which saves us a DOM API call)